newrelic/hooks/mlmodel_openai.py

```diff
@@ -93,25 +97,38 @@ def wrap_embedding_create(wrapped, instance, args, kwargs):
     if not response:
         return response
 
-    response_model = response.get("model", "")
-    response_usage = response.get("usage", {})
     response_headers = getattr(response, "_nr_response_headers", None)
+
+    # In v1, response objects are pydantic models so this function call converts the object back to a dictionary for backwards compatibility
+    if OPENAI_V1:
+        response = response.model_dump()
+
+    available_metadata = get_trace_linking_metadata()
+    span_id = available_metadata.get("span.id", "")
+    trace_id = available_metadata.get("trace.id", "")
+    embedding_id = str(uuid.uuid4())
+
     request_id = response_headers.get("x-request-id", "") if response_headers else ""
+    response_model = response.get("model", "")
+    response_usage = response.get("usage", {})
+    api_type = getattr(response, "api_type", "")
+    api_key = getattr(instance._client, "api_key", "") if OPENAI_V1 else getattr(response, "api_key", "")
```

Comment:
Let's move this up above to the rest of the openai logic. It looks like in v0 we were getting the key from openai not from the response so we should probably keep that the same.